client/components/services/useContactInfo.tsx

```diff
@@ -0,0 +1,35 @@
+import { useAuthContext } from '@contexts';
+import { useGrantApplication } from './useGrantApplicationContext';
+import { useState, useEffect } from 'react';
+import { API_ENDPOINT, AxiosPublic, REQUEST_METHOD } from 'constants/request-methods';
+import { useApplicationContext } from './useApplicationContext';
+
+export const useContactInfo = () => {
+  const keycloak = useAuthContext().keycloak;
+  const ApplicationContext = useApplicationContext();
+  const [initialValues, setInitialValues] = useState(ApplicationContext.initialValues.contactInfo);
+  const { updateProceedToNext } = useGrantApplication();
+
+  useEffect(() => {
+    setInitialValues(ApplicationContext.initialValues.contactInfo);
+  }, [ApplicationContext.initialValues.contactInfo]);
+
+  const handleSubmit = (values: any) => {
+    const patch = async () => {
+      const data = { contactInfo: values };
+
+      const options = {
+        method: REQUEST_METHOD.PATCH,
+        token: keycloak?.idToken,
+        data,
+        endpoint: API_ENDPOINT.applicationId(ApplicationContext.applicationId),
+      };
```

Comment:
Hey Arran, sorry, do you mean the ```endpoint:...``` part specifically or the whole ```options``` object? Thanks in advance!

Reply:
Sorry mostly the keycloak token, I'm used to the `axios.patch` format so I wasn't sure but the rest looks appropriate here

One thing to note, you'll often only have one service method per API endpoint, so it might not be entirely necessary to have them abstracted into a constants file.

ie: `createApplication` is tightly coupled to `axios.post(/application)` so that data being defined there could be appropriate rather then having one file that contains all endpoints. Defining the endpoint right in the service method also makes it a bit easier to understand/read what's going on at a glance without needing to look at multiple files

L25 here would make me stop and think, why am I calling a function for this endpoint? How does ApplicationContext play a role? When in reality it's as simple as `application/${applicationId}`